client/extensions/woocommerce/app/promotions/index.js

```diff
@@ -24,6 +26,23 @@ class Promotions extends Component {
 		} )
 	};
 
+	componentDidMount() {
+		const { site } = this.props;
+		if ( site && site.ID ) {
+			this.props.fetchCouponsPage( site.ID, 1 );
+		}
+	}
+
+	componentWillReceiveProps( newProps ) {
+		const { site } = this.props;
+		const newSiteId = newProps.site && newProps.site.ID || null;
```

Comment:
This and the next line could be simplified using `_.get` but fine as-is too.

Reply:
Yeah, I didn't figure it was worth adding another import just for one line.